### ISSUE TITLE
Cannot resolve servers. Missing data

### ISSUE THREAD

**Issue by ismriv** [h2non/resilient-consul#5]
@h2non, I'm still struggling to get this to work. I've added a service named `my-service`.

```
curl -X PUT -d '{"Datacenter": "vagrant-cloud", "Node": "my-node.example.com", "Address": "127.0.0.1", "Servi {"Service": "my-service", "Address": "127.0.0.1", "Port": 49000}}' http://127.0.0.1:8500/v1/catalog/register
```

When requesting it via CURL from Consul, I get: 

```
[
  {
    "Node": "my-node.example.com",
    "Address": "127.0.0.1",
    "ServiceID": "my-service",
    "ServiceName": "my-service",
    "ServiceTags": [

    ],
    "ServiceAddress": "127.0.0.1",
    "ServicePort": 49000,
    "ServiceEnableTagOverride": false,
    "CreateIndex": 216,
    "ModifyIndex": 216
  }
]
```

However, when I run the following:

```
var resilient = require('resilient');
var consul = require('resilient-consul');
var client = resilient();

client.use(consul({
  service: 'my-service',
  discoveryService: 'consul',
  datacenter: 'vagrant-cloud',
  enableSelfRefresh: true,
  servers: [
    'http://127.0.0.1:8500'
  ],
  onlyHealthy: false
}));

client.get('/', function (err, res) {
  console.log('Error:', err);
  console.log('Response:', res);
});
```

I get the error `Cannot resolve servers. Missing data`. I've debugged it for a while, and I see that when calling `hasValidServiceServers`, the function `resilient.servers()` is returning `{ servers: [], updated: 1453764937643 }`. However, I can see `handleResolution` being called for URL `http://172.0.0.1:8500/v1/catalog/service/my-service?_time=1453765689463&dc=vagrant-cloud` returning the service I added.

I've also tried adding a service to demo.consul.io, but I'm not allowed. Running the example (against demo.consul.io) from the README file throws `Discovery server response is invalid or empty`.

Any ideas?

**Comment by h2non** [comment]
For some reason the plugin is passing a bad servers list to resilient. Try adding a custom Consul response mapper function via mapServers option to ensure your passing the proper array of valid URLs.

**Comment by ismriv** [comment]
A custom `mapServers` function did the trick. The argument `list` passed is an array of service objects such as: 

```
[{
  "Node": "my-node.example.com",
  "Address": "127.0.0.1",
  "ServiceID": "my-service",
  "ServiceName": "my-service",
  "ServiceTags": [],
  "ServiceAddress": "127.0.0.1",
  "ServicePort": 49000,
  "ServiceEnableTagOverride": false,
  "CreateIndex": 216,
  "ModifyIndex": 216
}]
```

Resilient expects an array of URL strings, so it worked when adding the following custom mapping function:

```
mapServers: function (list) {
  return list.map(function (svc) {
    return 'http://' + svc.ServiceAddress + ':' + svc.ServicePort;
  });
}
```

However, the default function `mapServersFromCatalogEndpoint` seems to attempt doing this, but it will return `ServiceAddress` if found [instead of constructing the URI](https://github.com/h2non/resilient-consul/blob/master/consul.js#L100). Could this be a bug?

**Comment by h2non** [comment]
Consul addresses can be read in multiple was. In that case is not supperted properly, so there is required to map it properly as string. Feel free to do a PR to fix that.

**Comment by h2non** [comment]
Thanks for the PR. The feature was shipped in `0.1.5`. Feel free to update your dependency tree via `npm` or `bower`.

**Comment by ismriv** [comment]
Thanks @h2non for being so responsive!
